test/e2e/framework/framework.go

```diff
@@ -55,6 +56,8 @@ type Framework struct {
 	Clientset_1_2 *release_1_2.Clientset
 	Clientset_1_3 *release_1_3.Clientset
 
+	FederationClientset *federation_internalclientset.Clientset
+	// TODO: remove FederationClient, all the client access must be through FederationClientset
```

Comment:
It is definitely a line or two in the diff, but please also consider the additional debugging time required. I think getting federated service controller e2e working is the primary focus at the moment. Anything that takes even a few minutes extra without actually saving time elsewhere is out of question until that federated service e2es are merged.